website/src/theme/AnnouncementBar/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React from 'react';
+import clsx from 'clsx';
+import {useThemeConfig, useAnnouncementBar} from '@docusaurus/theme-common';
+import {translate} from '@docusaurus/Translate';
+import IconClose from '@theme/IconClose';
+import styles from './styles.module.css';
+export default function AnnouncementBar() {
+  const {isActive, close} = useAnnouncementBar();
+  const {announcementBar, announcementBarActive, announcementBarLink} = useThemeConfig();
+
+  if (!isActive || !announcementBarActive) {
+    return null;
+  }
+
+  const {content, backgroundColor, textColor, isCloseable} = announcementBar;
+
+  return (
+    <div
+      className={styles.announcementBar}
+      style={{
+        backgroundColor,
+        color: textColor,
+      }}
+      role="banner">
+      {isCloseable && <div className={styles.announcementBarPlaceholder} />}
+      {announcementBarLink ? (
+        <a 
```

Comment:
On the rendered site, it looks like we get `div.announcementBarContent_dpRF` inside the new `a` you set up. Is there any reason we couldn't collapse that into a single `a` tag? I tend to cut out unnecessary markup as much as possible. ("It would be a pain to refactor" is a very good reason -- I don't want you to have to swizzle `AnnouncementBarContent` as well!)

Reply:
My main reason for keeping it as-is here is i'm not 100% sure if Docusaurus is depending on the `announcementBarContent` being a `div` under the hood anywhere. If any JS they are using for the banner functionality is looking for a `div`, and I change it to an anchor, it could potentially break it.

So what I did here was broke out the `announcementBarContent` div into it's own component, and simply wrapping that component with an anchor if `announcementBarLink` is set in the config. That way none of the existing setup is altered, and just adding additional functionality for the new `announcementBarLink` config option I added.

Docusaurus doesn't support having the entire banner as a link, and expects users to add links inline with the text, such as their example in the docs. This doesn't match our functionality on WWW where the whole banner is clickable, which was the reason for swizzling the component and wrapping the content with an anchor.
```
announcementBar: {
      id: 'support_us',
      content:
        'We are looking to revamp our docs, please fill <a target="_blank" rel="noopener noreferrer" href="#">this survey</a>',
      backgroundColor: '#fafbfc',
      textColor: '#091E42',
      isCloseable: false,
    },
```

Reply:
Ah. I figured it was More Complicated than it looked!